### PR TITLE
Add angle between vectors functions + Fix lookAt and lookTowards tests

### PR DESCRIPTION
**Pull request by ThomasGorisse**

### CODE REVIEW

**Review by romainguy — CHANGES_REQUESTED** [review]
Good catch, but this contains other changes as well.

File: src/commonTest/kotlin/dev/romainguy/kotlin/math/MatrixTest.kt [hunk 1]
```diff
@@ -523,6 +523,12 @@ class MatrixTest {
             Float4(4f, 8f, 12f, 16f)
         )
 
+        internal fun assertFloatEquals(expected: Float, actual: Float, delta: Float = 0.0001f) {
```
Comment (romainguy) [inline src/commonTest/kotlin/dev/romainguy/kotlin/math/MatrixTest.kt:526]: It's not required, you can use `assertEquals(expected, actual, absoluteTolerance)` from `kotlin.test`.

**Review by romainguy — CHANGES_REQUESTED** [review]

**Comment by ThomasGorisse** [comment]
Sorry for the friday evenining PR branch switch.

**Comment by ThomasGorisse** [comment]
Just to let you know, we need this for the actual integration of Filament Pick+Manipulator.
It's a little bit tricky to handle the camera manipulator in one case and the model transform in the other (AR)
https://github.com/SceneView/sceneview-android/pull/38

File: src/commonTest/kotlin/dev/romainguy/kotlin/math/MatrixTest.kt (hunk 1 above)
Comment (ThomasGorisse) [reply src/commonTest/kotlin/dev/romainguy/kotlin/math/MatrixTest.kt]: Fix [f883ae1](https://github.com/romainguy/kotlin-math/pull/46/commits/f883ae15a21bb2ea4e2df20ceaf8272089c89bf6)

**Review by romainguy — APPROVED** [review]

**Comment by romainguy** [comment]
Thanks!

**Pull request merged**